etc/initdev/Vagrantfile

```diff
@@ -4,7 +4,7 @@
 VAGRANTFILE_API_VERSION = "2"
 
 Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
-  config.vm.box = "ubuntu/vivid64" # 15.04
+  config.vm.box = "bento/ubuntu-16.04"
```

Comment:
I guest it's just the existence. Tried host and docker first without luck in the first time. So turned to Vagrant, and realized it was just empty ubuntu + go.


Adding a dev guide section in the documentation will definitely help.
 
Another thing might help is to break up the Makefile a little, as it now has more than 60+ targets. Or at least reorder the targets so recommended flow will be spotted more easily.

For example, this morning I tried the following
`
make
make build
make deps; make build
make proto
`

Then I saw the Dockerfile(s) in the root. So
`
docker build -t pachyderm
docker run -it pachyderm
make deps; make build;
...
`
Still no luck. So I turned to slack for help.

However, if I just keep scrolling down the Makefile a coupe of pages, I might have found the docker-build and found it earlier.